jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3181,7 +3241,20 @@ public Method getEnclosingMethod() throws SecurityException {
 public Class<?> getEnclosingClass() throws SecurityException {
 	Class<?> enclosingClass = getDeclaringClass();
 	if (enclosingClass == null) {
-		enclosingClass = getEnclosingObjectClass();
+		if (cachedEnclosingClassOffset == -1) {
+			cachedEnclosingClassOffset = getFieldOffset("cachedEnclosingClass"); //$NON-NLS-1$
+		}
+		if (cachedEnclosingClass == null) {
+			Class<?> localEnclosingClass = getEnclosingObjectClass();
+			if (localEnclosingClass == null)
```

Comment:
Can you add the `{ }` to this if statement?

Reply:
added `{ }` as suggested. thanks!